Integration test build failure due to arrow-rs using unstable feature

**Describe the bug**
While running the integration test in dev/, the docker build failed due to using of unstable feature:
```
#29 293.2 error[E0658]: use of unstable library feature 'partition_point': new API
#29 293.2   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/arrow-4.3.0/src/compute/kernels/partition.rs:88:14
#29 293.2    |
#29 293.2 88 |             .partition_point(|idx| {
#29 293.2    |              ^^^^^^^^^^^^^^^
#29 293.2    |
#29 293.2    = note: see issue #73831 <https://github.com/rust-lang/rust/issues/73831> for more information
````

**To Reproduce**
run ./dev/integration-tests.sh on master branch

**Expected behavior**
Integration test run as expected

--- a/dev/docker/ballista-base.dockerfile
+++ b/dev/docker/ballista-base.dockerfile
@@ -96,4 +96,6 @@ RUN cargo install cargo-build-deps
 
 # prepare toolchain
 RUN rustup update && \
+    rustup toolchain install nightly-2021-05-10 && \
+    rustup default nightly-2021-05-10 && \
     rustup component add rustfmt

--- a/dev/docker/ballista-base.dockerfile
+++ b/dev/docker/ballista-base.dockerfile
@@ -23,7 +23,7 @@
 
 
 # Base image extends debian:buster-slim
-FROM rust:1.51.0-buster AS builder
+FROM rust:1.53.0-buster AS builder
 
 RUN apt update && apt -y install musl musl-dev musl-tools libssl-dev openssl
 
@@ -96,6 +96,4 @@ RUN cargo install cargo-build-deps
 
 # prepare toolchain
 RUN rustup update && \
-    rustup toolchain install nightly-2021-05-10 && \
-    rustup default nightly-2021-05-10 && \
     rustup component add rustfmt

